depends/packages/zeromq.mk

```diff
@@ -3,7 +3,7 @@ $(package)_version=4.2.3
 $(package)_download_path=https://github.com/zeromq/libzmq/releases/download/v$($(package)_version)/
 $(package)_file_name=$(package)-$($(package)_version).tar.gz
 $(package)_sha256_hash=8f1e2b2aade4dbfde98d82366d61baef2f62e812530160d2e6d0a5bb24e40bc0
-$(package)_patches=0001-fix-build-with-older-mingw64.patch 0002-disable-pthread_set_name_np.patch
+$(package)_patches=0001-fix-build-with-older-mingw64.patch 0002-disable-pthread_set_name_np.patch 0003-new_configure.patch
```

Comment:
Could we [copy our files](https://github.com/bitcoin/bitcoin/blob/master/depends/packages/bdb.mk#L18) instead, since we have to keep ours up to date anyway?

Reply:
I've added some commits to a branch that attempt to do that: https://github.com/fanquake/bitcoin/tree/riscv-depends-no-zmq-patching. It also updates the depends config.guess/sub to latest upstream.

@laanwj You can cherry pick from there if you'd like. Note that branch doesn't include re-running `./autogen.sh` like below. I'll comment about that shortly.